src/main/java/com/aws/greengrass/mqttclient/IotCoreTopicValidator.java

```diff
@@ -141,19 +153,12 @@ private static Optional<String> removePrefix(String topic, String prefixRegex) {
         return Optional.empty();
     }
 
-    /**
-     * Extract the prefix from a topic, with n number of topic parts.
-     *
-     * <p>Example: prefix("a/b/c/d", 2) -> "a/b"
-     *
-     * @param topic  non-null, non-empty, trimmed topic
-     * @param numParts number of topic parts to include in the result
-     * @return topic prefix
-     */
-    private static String prefix(String topic, int numParts) {
-        String[] parts = topic.split("/");
-        return Arrays.stream(parts)
-                .limit(Math.min(parts.length, numParts))
-                .collect(Collectors.joining("/"));
+    private static boolean containsWildcards(String topic) {
+        for (char c : topic.toCharArray()) {
```

Comment:
this creates a copy of the underlying array, seems probably more expensive than just doing a contains check twice